scripts: improve dl_cleanup script

Ansuel opened the pull request:
Improve dl_cleanup by adding an option to also clean the build directory related to the downloaded package.
Also add some check to the existence of the provided dl and build directory.

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

oldium commented on scripts/dl_cleanup.py:
```diff
@@ -90,6 +98,7 @@ def parseVer_GIT(match, filepath):
 
 versionRegex = (
 	(re.compile(r"(.+)[-_](\d+)\.(\d+)\.(\d+)\.(\d+)"), parseVer_1234),	# xxx-1.2.3.4
+	(re.compile(r"(.+)[-_](\d\d\d\d)-?(\d\d)-?(\d\d)-"), parseVer_ymd),	# xxx-YYYY-MM-DD-GIT_SHASUM
```
Shouldn't this reference the new function parseVer_ymd_GIT_SHASUM?

Ansuel replied:
yes it's a typo

aparcar commented:
Would you mind splitting this PR into multiple commits?

Ansuel commented:
Sure will do that tomorrow

Ansuel commented:
@aparcar WTH 15 days O_O
SORRY.....  (exam session is killing me...)

Anyway i split this in 3 more useful commits

aparcar commented on scripts/dl_cleanup.py:
```diff
@@ -90,6 +97,7 @@ def parseVer_GIT(match, filepath):
 
 versionRegex = (
 	(re.compile(r"(.+)[-_](\d+)\.(\d+)\.(\d+)\.(\d+)"), parseVer_1234),	# xxx-1.2.3.4
+	(re.compile(r"(.+)[-_](\d\d\d\d)-?(\d\d)-?(\d\d)-"), parseVer_ymd_GIT_SHASUM),	# xxx-YYYY-MM-DD-GIT_SHASUM
```
Cool, do you have an example package I could test this with?

Ansuel replied:
Ok found one example why this is required
Example package is
openwrt-keyring

```
Without the additional regex the version detected is
expat 36894051140322197504
openwrt-keyring-2021-02-20 90905565772833481555968
libubox 37280870617400523882496
With the additional regex
expat 36894051140322197504
openwrt-keyring 37280870336002856583168
libubox 37280870617400523882496
```

This cause a problem in detecting the version of the package as the package is detected as a separate one with the date.

Ansuel replied:
Think i should add this to the commit description?

aparcar commented:
Thanks, I merged your second second commit. For the other commits, please limit the number of characters to 80 per line.

While cleaning the script, tabs in Python is somewhat awkward and should be replaced. Mind applying the `black` code style?

Also we could assume both `dl/` as download folder and `build_dir` as build folder, don't you think? This could be another commit if you feel like it.

Ansuel committed:
scripts: assume dl/ the default dir for dl_cleanup script

Assume dl/ the default dl dir and make it configurable if someone have that in
a different place.

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

Ansuel committed:
scripts: add clean of build dir to dl_cleanup script

Improve dl_cleanup by adding an option to also clean the build directory
related to the downloaded package.
The script will check every directory in build_dir/ and check if any
old package is present there. If outdated package are found, the old
one are cleared leaving only the last one.

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

Ansuel commented:
@aparcar i improved this even more...
- Converted to the new style.
- Set dl and build_dir by default
- Also now the function search in every directory in build_dir... (with the old implementation you had to define the build_dir with the target path... now it does check every dir there, much better. Thx for the hint about assuming build_dit)

Ansuel commented:
I also added some extra code to handle linux and gcc package.

Ansuel commented:
@aparcar news on this?

aparcar commented on scripts/dl_cleanup.py:
```diff
@@ -191,25 +194,20 @@ def main(argv):
     except getopt.GetoptError as e:
         usage()
         return 1
-    directory = args[0]
```
To be compatible with the buildbot we should add this legacy behaviour. Meaning, if `-D` is not defined assume `arg[0]`. What do you think?

Ansuel replied:
buildbot use this script?

so the problem is that buildbot would still provide the args and would cause some problem? 
I think that's the only way to solve this. Correct?

aparcar commented:
Please rebase this PR

Ansuel commented:
@aparcar will rebase as soon as we take a decision on the legacy problem with buildbot

Ansuel commented:
@aparcar rebased on top of master.

Ansuel commented:
Wow this is old. Any news?

Ansuel committed:
scripts: add additional regex for dl_cleanup

Add additional regex for dl_cleanup script to handle
case with xxx-v1.2a and xxx-v1.2.3a

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

aparcar commented:
Workflow wise, you rebase it, push it again to the feature branch, merge it locally and then push it to master?

Ansuel commented:
@aparcar yes this is the only way to flag a pr as merged saddly. I created a script in maintainer-tools to make this automatic.
Should we skip checks when a particular tag is set?
